pkg/component/gardeneradmissioncontroller/admissioncontroller_test.go

```diff
@@ -35,31 +72,1101 @@ var _ = Describe("AdmissionController", func() {
+							{Type: resourcesv1alpha1.ResourcesHealthy, Status: gardencorev1beta1.ConditionTrue},
+						},
+					},
+				}
+				Expect(fakeClient.Create(ctx, runtimeManagedResource)).To(Succeed())
+
+				err := deployer.WaitCleanup(ctx)
+
+				multiErr, ok := err.(*multierror.Error)
+				Expect(ok).To(BeTrue())
+				Expect(multiErr.Errors).To(HaveLen(1))
+				Expect(multiErr.Errors[0]).To(MatchError("retry failed with max attempts reached, last error: resource some-namespace/gardener-admission-controller-runtime still exists"))
+			})
+		})
+
+		Context("when Virtual ManagedResource is still available", func() {
+			It("should time out waiting", func() {
+				runtimeManagedResource := &resourcesv1alpha1.ManagedResource{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      managedResourceNameVirtual,
+						Namespace: namespace,
+					},
+					Status: resourcesv1alpha1.ManagedResourceStatus{
+						Conditions: []gardencorev1beta1.Condition{
+							{Type: resourcesv1alpha1.ResourcesApplied, Status: gardencorev1beta1.ConditionTrue},
+							{Type: resourcesv1alpha1.ResourcesHealthy, Status: gardencorev1beta1.ConditionTrue},
+						},
+					},
+				}
+				Expect(fakeClient.Create(ctx, runtimeManagedResource)).To(Succeed())
+
+				err := deployer.WaitCleanup(ctx)
+
+				multiErr, ok := err.(*multierror.Error)
+				Expect(ok).To(BeTrue())
+				Expect(multiErr.Errors).To(HaveLen(1))
+				Expect(multiErr.Errors[0]).To(MatchError("retry failed with max attempts reached, last error: resource some-namespace/gardener-admission-controller-virtual still exists"))
+			})
 		})
 	})
 
 	Describe("#Destroy", func() {
+		Context("when resources don't exist", func() {
+			It("should successful destroy", func() {
+				Expect(deployer.Destroy(ctx)).To(Succeed())
+
+				verifyResourcesGone(ctx, fakeClient, namespace)
+			})
+		})
+
 		It("should successful destroy", func() {
+			runtimeManagedResource := &resourcesv1alpha1.ManagedResource{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      managedResourceNameRuntime,
+					Namespace: namespace,
+				},
+				Spec: resourcesv1alpha1.ManagedResourceSpec{
+					SecretRefs: []corev1.LocalObjectReference{{Name: "managedresource-" + managedResourceNameRuntime}},
+				},
+			}
+			runtimeManagedResourceSecret := &corev1.Secret{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      runtimeManagedResource.Spec.SecretRefs[0].Name,
+					Namespace: namespace,
+				},
+			}
+
+			Expect(fakeClient.Create(ctx, runtimeManagedResourceSecret)).To(Succeed())
+			Expect(fakeClient.Create(ctx, runtimeManagedResource)).To(Succeed())
+
 			Expect(deployer.Destroy(ctx)).To(Succeed())
+
+			verifyResourcesGone(ctx, fakeClient, namespace)
 		})
 	})
 })
+
+func verifyResourcesGone(ctx context.Context, fakeClient client.Client, namespace string) {
+	Expect(fakeClient.Get(ctx, client.ObjectKey{Namespace: namespace, Name: "managedresource-" + managedResourceNameRuntime}, &corev1.Secret{})).To(matchers.BeNotFoundError())
```

Comment:
I'm fine with `ExpectWithOffset` for the `verifyResourcesGone` function since it's used directly with other expectations (see [here](https://github.com/gardener/gardener/blob/32ec5e0ab1fb46ec748e68a78ee8d918f7b1faa4/pkg/component/gardeneradmissioncontroller/admissioncontroller_test.go#L412-L417)).

However, for `verifyExpectations` an `ExpectWithOffset` rather causes a confusing output:

```
• [FAILED] [0.027 seconds]
GardenerAdmissionController #Deploy with common values [It] should successfully deploy

[FAILED] Expected
      <string>: "...rdener-admission-controller-930189d2
        namespace: some-namespace
      "
  to equal                                                            |
      <string>: "...rdener-admission-controller-930189d2
        namespace: kube-system
      "
  In [It] at: pkg/component/gardeneradmissioncontroller/gardener_admission_controller_test.go:487 @ 07/19/23 12:07:07.939
```

--> Stack link points directly to ManagedResource expectation that fails 🎯 

```
• [FAILED] [0.027 seconds]
GardenerAdmissionController #Deploy with common values [It] should successfully deploy

Timeline >>
[FAILED] Expected
      <string>: "...rdener-admission-controller-930189d2
        namespace: some-namespace
      "
  to equal                                                            |
      <string>: "...rdener-admission-controller-930189d2
        namespace: kube-system
      "
  In [It] at: pkg/component/gardeneradmissioncontroller/gardener_admission_controller_test.go:143
```

--> Stack link points to `It()`, i.e. you don't know with expectation fails 🤷‍♂️. The test name is printed anyway, so there is no benefit in pointing to the `It()`, IMO.